coretex/utils/docker.py

```diff
@@ -147,3 +150,30 @@ def getResourceLimits() -> Tuple[int, int]:
     jsonOutput = json.loads(output)
 
     return jsonOutput["NCPU"], round(jsonOutput["MemTotal"] / (1024 ** 3))
+
+
+def getDockerConfigPath() -> Optional[Path]:
+    if platform.system() == "Darwin":
+        return Path.home().joinpath("Library", "Group Containers", "group.com.docker", "settings.json")
+    elif platform.system() == "Windows":
+        return Path.home().joinpath("AppData", "Roaming", "Docker", "settings.json")
+    elif platform.system() == "Linux":
+        return Path.home().joinpath(".docker", "desktop", "settings.json")
+    else:
+        return None
+
+
+def getDockerSwapLimit() -> int:
+    configPath = getDockerConfigPath()
+
+    if configPath is None:
+        return getTotalSwapMemory()
+
+    with configPath.open("r") as configFile:
+        configJson = json.load(configFile)
+
+    swapLimit = configJson.get("swapMiB")
+    if not isinstance(swapLimit, int):
```

Comment:
if `swapLimit` is not valid do not raise an error, return either 0, or if it will work return the value of swap configured on the system.